smr.js

```diff
@@ -31,17 +28,17 @@ MatrixProduct.prototype.push = MatrixProduct.prototype.addRowAndColumn
 
 function Regression(options) {
 
-  if(!options)
-    throw new Error('missing options')
+  // if(!options)
+  //   throw new Error('missing options')
 
-  if(!('numX' in options))
-    throw new Error(
-      'you must give the width of the X dimension as the property numX')
+  // if(!('numX' in options))
+  //   throw new Error(
+  //     'you must give the width of the X dimension as the property numX')
 
 
-  if(!('numY' in options))
-    throw new Error(
-      'you must give the width of the X dimension as the property numY')
+  // if(!('numY' in options))
+  //   throw new Error(
+  //     'you must give the width of the X dimension as the property numY')
```

Comment:
@RyanMarcus any thoughts on this? @oebspm is repoting performance improvements if we remove these checks.  I'm inclined to do so since performance is one of the main concerns of this library.